aws-lc-rs/src/unstable/kdf/sskdf.rs

```diff
@@ -0,0 +1,245 @@
+    id: SskdfAlgorithmId::HmacSha256,
+    digest: AlgorithmID::SHA256,
+};
+
+/// SSKDF with HMAC-SHA384
+pub const SSKDF_HMAC_SHA384: SskdfHmacAlgorithm = SskdfHmacAlgorithm {
+    id: SskdfAlgorithmId::HmacSha384,
+    digest: AlgorithmID::SHA384,
+};
+
+/// SSKDF with HMAC-SHA512
+pub const SSKDF_HMAC_SHA512: SskdfHmacAlgorithm = SskdfHmacAlgorithm {
+    id: SskdfAlgorithmId::HmacSha512,
+    digest: AlgorithmID::SHA512,
+};
+
+/// SSKDF with SHA1
+pub const SSKDF_DIGEST_SHA1: SskdfDigestAlgorithm = SskdfDigestAlgorithm {
+    id: SskdfAlgorithmId::DigestSha1,
+    digest: AlgorithmID::SHA1,
+};
+
+/// SSKDF with SHA224
+pub const SSKDF_DIGEST_SHA224: SskdfDigestAlgorithm = SskdfDigestAlgorithm {
+    id: SskdfAlgorithmId::DigestSha224,
+    digest: AlgorithmID::SHA224,
+};
+
+/// SSKDF with SHA256
+pub const SSKDF_DIGEST_SHA256: SskdfDigestAlgorithm = SskdfDigestAlgorithm {
+    id: SskdfAlgorithmId::DigestSha256,
+    digest: AlgorithmID::SHA256,
+};
+
+/// SSKDF with SHA384
+pub const SSKDF_DIGEST_SHA384: SskdfDigestAlgorithm = SskdfDigestAlgorithm {
+    id: SskdfAlgorithmId::DigestSha384,
+    digest: AlgorithmID::SHA384,
+};
+
+/// SSKDF with SHA512
+pub const SSKDF_DIGEST_SHA512: SskdfDigestAlgorithm = SskdfDigestAlgorithm {
+    id: SskdfAlgorithmId::DigestSha512,
+    digest: AlgorithmID::SHA512,
+};
+
+/// SSKDF algorithm using HMAC
+pub struct SskdfHmacAlgorithm {
+    id: SskdfAlgorithmId,
+    digest: AlgorithmID,
+}
+
+impl SskdfHmacAlgorithm {
+    /// Returns the SSKDF Algorithm Identifier
+    #[must_use]
+    pub fn id(&self) -> SskdfAlgorithmId {
+        self.id
+    }
+}
+
+impl core::fmt::Debug for SskdfHmacAlgorithm {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        core::fmt::Debug::fmt(&self.id, f)
+    }
+}
+
+/// SSKDF algorithm using digest
+pub struct SskdfDigestAlgorithm {
+    id: SskdfAlgorithmId,
+    digest: AlgorithmID,
+}
+
+impl SskdfDigestAlgorithm {
+    /// Returns the SSKDF Algorithm Identifier
+    #[must_use]
+    pub fn id(&self) -> SskdfAlgorithmId {
+        self.id
+    }
+}
+
+impl core::fmt::Debug for SskdfDigestAlgorithm {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        core::fmt::Debug::fmt(&self.id, f)
+    }
+}
+
+/// Single-step (One-step) Key Derivation Function Algorithm Identifier
+#[non_exhaustive]
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub enum SskdfAlgorithmId {
+    /// SSKDF with SHA1
+    DigestSha1,
+
+    /// SSKDF with SHA224
+    DigestSha224,
+
+    /// SSKDF with SHA256
+    DigestSha256,
+
+    /// SSKDF with SHA384
+    DigestSha384,
+
+    /// SSKDF with SHA512
+    DigestSha512,
+
+    /// SSKDF with HMAC-SHA1
+    HmacSha1,
+
+    /// SSKDF with HMAC-SHA224
+    HmacSha224,
+
+    /// SSKDF with HMAC-SHA256
+    HmacSha256,
+
+    /// SSKDF with HMAC-SHA384
+    HmacSha384,
+
+    /// SSKDF with HMAC-SHA512
+    HmacSha512,
+}
```

Comment:
Since `sskdf_hmac` and `sskdf_digest` are separate functions, I think it would likewise be better to have separate `SskdfHmacAlgorithmId` and `SskdfDigestAlgorithmId`.